node/core/approval-voting/src/lib.rs

```diff
@@ -539,6 +533,95 @@ struct ApprovalStatus {
 	block_tick: Tick,
 }
 
+enum ApprovalOutcome {
+	Approved,
+	Failed(InvalidCandidate),
+}
+
+struct ApprovalState {
+	validator_index: ValidatorIndex,
+	candidate_hash: CandidateHash,
+	candidate_index: CandidateIndex,
+	status: ApprovalOutcome,
+}
+
+impl ApprovalState {
+	fn approved(
+		validator_index: ValidatorIndex,
+		candidate_hash: CandidateHash,
+		candidate_index: CandidateIndex
```

Comment:
style: multi-line lists should end with commas. we don't use rustfmt so you should do this manually
```suggestion
		candidate_index: CandidateIndex,
```

Reply:
here and other places.